contracts/src/v0.8/l2ep/dev/scroll/ScrollSequencerUptimeFeed.sol

```diff
@@ -0,0 +1,242 @@
+  /// @param roundId The round ID to record
+  /// @param status Sequencer status
+  /// @param timestamp The L1 block timestamp of status update
+  function _recordRound(uint80 roundId, bool status, uint64 timestamp) private {
+    uint64 updatedAt = uint64(block.timestamp);
+    Round memory nextRound = Round(status, timestamp, updatedAt);
+    FeedState memory feedState = FeedState(roundId, status, timestamp, updatedAt);
+
+    s_rounds[roundId] = nextRound;
+    s_feedState = feedState;
+
+    emit NewRound(roundId, msg.sender, timestamp);
+    emit AnswerUpdated(_getStatusAnswer(status), roundId, timestamp);
+  }
+
+  /// @notice Helper function to update when a round was last updated
+  /// @param roundId The round ID to update
+  /// @param status Sequencer status
+  function _updateRound(uint80 roundId, bool status) private {
+    uint64 updatedAt = uint64(block.timestamp);
+    s_rounds[roundId].updatedAt = updatedAt;
+    s_feedState.updatedAt = updatedAt;
+    emit RoundUpdated(_getStatusAnswer(status), updatedAt);
+  }
+
+  /// @notice Record a new status and timestamp if it has changed since the last round.
+  /// @dev This function will revert if not called from `l1Sender` via the L1->L2 messenger.
+  ///
+  /// @param status Sequencer status
+  /// @param timestamp Block timestamp of status update
+  function updateStatus(bool status, uint64 timestamp) external override {
+    FeedState memory feedState = s_feedState;
+    if (
+      msg.sender != address(s_l2CrossDomainMessenger) || s_l2CrossDomainMessenger.xDomainMessageSender() != s_l1Sender
+    ) {
+      revert InvalidSender();
+    }
+
+    // Ignore if latest recorded timestamp is newer
+    if (feedState.startedAt > timestamp) {
+      emit UpdateIgnored(feedState.latestStatus, feedState.startedAt, status, timestamp);
+      return;
+    }
+
+    if (feedState.latestStatus == status) {
+      _updateRound(feedState.latestRoundId, status);
+    } else {
+      feedState.latestRoundId += 1;
+      _recordRound(feedState.latestRoundId, status, timestamp);
+    }
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestAnswer() external view override checkAccess returns (int256) {
+    FeedState memory feedState = s_feedState;
+    return _getStatusAnswer(feedState.latestStatus);
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestTimestamp() external view override checkAccess returns (uint256) {
+    FeedState memory feedState = s_feedState;
+    return feedState.startedAt;
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function latestRound() external view override checkAccess returns (uint256) {
+    FeedState memory feedState = s_feedState;
+    return feedState.latestRoundId;
+  }
+
+  /// @inheritdoc AggregatorInterface
+  function getAnswer(uint256 roundId) external view override checkAccess returns (int256) {
+    if (_isValidRound(roundId)) {
+      return _getStatusAnswer(s_rounds[uint80(roundId)].status);
```

Comment:
Could we invert this if so we have the error case in the if and the happy path at the end of the function? Same for getTimestamp